File: Programmars/Level_2/타겟 넘버.py
def DFS(level, total):
    global answer
    if level == len(nums):
        if total == t:
            answer += 1
    else:
        DFS(level+1, total + nums[level])
        DFS(level+1, total - nums[level])


def solution(numbers, target):
    global answer, nums, t, result
    answer = 0
    nums = numbers
    t = target
    result = []
    DFS(0, 0)
    return answer


print(solution([1, 1, 1, 1, 1], 3))  # 5

File: JavaScript/Programmars/깊이, 너비 우선 탐색/타겟 넘버.js
function solution(numbers, target) {
  let answer = 0;

  function DFS(level, total) {
    if (level < numbers.length) {
      DFS(level + 1, total + numbers[level]);
      DFS(level + 1, total - numbers[level]);
    } else {
      if (total === target) answer++;
    }
  }

  DFS(0, 0);

  return answer;
}

console.log(solution([1, 1, 1, 1, 1], 3)); // 5
console.log(solution([4, 1, 2, 1], 4)); // 2
